File: src/main/java/com/devdaily/dbgrinder/model/JavaBeanFactory.java
package com.devdaily.dbgrinder.model;

import java.io.IOException;

/**
 * A simple Factory that should produce JavaBeans and
 * JavaBean components.
 */
public class JavaBeanFactory
{
  private static String indentation = "  ";
  public static String INDENT_TWO_CHARACTERS = "  ";
  public static String INDENT_ONE_TAB = "\t";

  static String writeJavaBeanGetMethod(ColumnData cd, String newName, String varName)
  throws IOException
  {
    String s = "\n";
    s += getIndentation() + "public " + cd.getJavaType() + " get" + newName + "()\n";
    s += getIndentation() + "{\n";
    s += getIndentation() + getIndentation() + "return " + varName + ";\n";
    s += getIndentation() + "}";
    return s;
  }

  static String writeJavaBeanSetMethod(ColumnData cd, String methodReturnType, String newName, String varName)
  throws IOException
  {
    String s = "\n";
    s += getIndentation() +  "public " + methodReturnType + " set" + newName + "(" + cd.getJavaType() + " " + varName + ")\n";
    s += getIndentation() +  "{\n";
    s += getIndentation() +  getIndentation() + "this." + varName + " = " + varName + ";\n";
    s += getIndentation() +  "}";
    return s;
  }

  public static void setIndentation(String newIndentation)
  {
    indentation = newIndentation;
  }

  public static String getIndentation()
  {
    return indentation;
  }

  public static String getIndentation(int numTimes)
  {
    String s = "";
    for (int i=0; i<numTimes; i++)
    {
      s += getIndentation();
    }
    return s;
  }

}

File: src/main/scala/com/alvinalexander/cato/model/JavaBeanFactory.scala
package com.alvinalexander.cato.model

import com.devdaily.dbgrinder.model.ColumnData

/**
 * A simple Factory that should produce JavaBeans and
 * JavaBean components.
 */
object JavaBeanFactory {

    var indentation = "  "
    val INDENT_TWO_CHARACTERS = "  "
    val INDENT_ONE_TAB = "\t"
  
    // TODO wrap in try
    def writeJavaBeanGetMethod(cd: ColumnData, newName: String, varName: String): String = {
        var s = "\n"
        s += getIndentation + "public " + cd.getJavaType + " get" + newName + "\n"
        s += getIndentation + "{\n"
        s += getIndentation + getIndentation + "return " + varName + ";\n"
        s += getIndentation + "}"
        s
    }
  
    // TODO wrap in try
    def writeJavaBeanSetMethod(cd: ColumnData, methodReturnType: String, newName: String, varName: String) {
        var s = "\n"
        s += getIndentation +  "public " + methodReturnType + " set" + newName + "(" + cd.getJavaType + " " + varName + ")\n"
        s += getIndentation +  "{\n"
        s += getIndentation +  getIndentation + "this." + varName + " = " + varName + ";\n"
        s += getIndentation +  "}"
        return s;
    }
  
    def setIndentation(newIndentation: String) {
        indentation = newIndentation
    }
  
    def getIndentation = indentation
  
    def getIndentation(numTimes: Int) {
        var s = ""
        for (i <- 0 to numTimes) {
            s += getIndentation
        }
        s
    }

}